.github/actions/build_with_cache/action.yml

```diff
@@ -2,17 +2,36 @@ name: build_with_cache
 description: builds the source code if cache miss and caches the result
 inputs:
   node-version:
-    default: 18
+    required: true
+  cdk-version:
+    required: true
```

Comment:
When I was debugging this. I discovered that we have a mess.
I.e. tests claiming that they run on Node 20 were either using wrong node (18) or wrong install/build cache.

Therefore I made these parameters mandatory in couple of places to assert correctness. And I added runtime validation (it turns out that `required` matters only for intellisense really...).